tools/psm_validation/psm_validation.xml

```diff
@@ -0,0 +1,56 @@
+    <tool id="psmvalidator" name="PSM-Fragment Validator" version="@VERSION@">
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <description>Validate PSMs against Ion Fragmentation</description>
+    <expand macro="pkg_requirement" />
+    <command detect_errors="exit_code">
+        <![CDATA[
+        python '$__tool_directory__/application.py' 
+            -d '$mz_db_name'
+            -p '$peptide_sequences'
+            #if $neutral_loss_ions:
+                -n
+            #end if
+            #if $internal_ions:
+                -i
+            #end if
+            -b $b_ion_run
+            -y $y_ion_run
+            -e $epsilon
+            #if $b_ion_run < 0:
+                --test
+            #end if
+        ]]>
+    </command>
+    <inputs>
+        <param name="mz_db_name" type="data" format="mz.sqlite" label="MZSQLite DB" />
+        <param name="peptide_sequences" type="data" format="tabular" label="Peptide Sequence List" />
+        <param name="b_ion_run" type="integer" value="2" label="Consecutive B-Ions" />
+        <param name="y_ion_run" type="integer" value="2" label="Consecutive Y-Ions" />
+        <param name="neutral_loss_ions" type="boolean" checked="True" label="Use Neutral Loss" />
+        <param name="internal_ions" type="boolean" checked="True" label="Use Internals" />
+        <param name="epsilon" type="float" label="Epsilon">0.5</param>
+    </inputs>
+    <outputs>
+        <data name="html_output" format="html" from_work_dir="index.html" label="PSM Ion Fragementation Report" />
+        <data name="text_output" format="csv" from_work_dir="results.csv" label="CSV Results" />
```

Comment:
@tmcgowan can we make sure this is a tabular formatted file? `tabular`?

Reply:
The outputs are HTML and CSV. Neither is tabular. I am sorry, but I do not understand your request.

Reply:
something like that `sed -e 's/,/\t/g' /tmp/file.csv > /tmp/file.tabbed` after the tools has run ... Galaxy has 100 of tools capable of handling tabular files - for csv we need to convert them in a separate step to tabular to make use of those tools. This is the reason I suggerst to let your tool spit out a tabular file natively. does that make sense?

Reply:
OK, now I understand.